ios/RNGestureHandlerManager.m

```diff
@@ -132,25 +128,35 @@ - (void)handleClearJSResponder
 
 #pragma mark Root Views Management
 
-- (void)registerRootViewIfNeeded:(UIView*)childView
+- (void)registerRootViewGestureRecognizerIfNeeded:(UIView*)childView
 {
     UIView *parent = childView;
-    while (parent != nil && ![parent isKindOfClass:[RCTRootView class]]) parent = parent.superview;
-    
-    RCTRootView *rootView = (RCTRootView *)parent;
-    UIView *rootContentView = rootView.contentView;
-    if (rootContentView != nil && ![_rootViews containsObject:rootContentView]) {
-        RCTLifecycleLog(@"[GESTURE HANDLER] Initialize gesture handler for root view %@", rootContentView);
-        [_rootViews addObject:rootContentView];
-        RNRootViewGestureRecognizer *recognizer = [RNRootViewGestureRecognizer new];
-        recognizer.delegate = self;
-        rootContentView.userInteractionEnabled = YES;
-        [rootContentView addGestureRecognizer:recognizer];
+    while (parent != nil && ![parent respondsToSelector:@selector(touchHandler)]) parent = parent.superview;
+
+    UIView *rootView = [[parent performSelector:@selector(touchHandler)] view];
+  
+    if (rootView == nil) {
+      return;
     }
+  
+    for (RNRootViewGestureRecognizer *recognizer in _rootViewGestureRecognizers) {
```

Comment:
doesn't look like we really need to check this table here, it should be enough to check if `view.gestureRecognizers` contains an object of tyle `RNRootViewGestureRecognizer` – this should be a simpler check as we wouldn't need to iterate over hash table